In [1]:
import pandas as pd
import numpy as np
import os
from pathlib import Path

In [61]:
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all"

## Data ingest

In [57]:
#function to load data
def load_data(filename) -> pd.DataFrame:
    dir = Path.cwd().parent
    path = dir/'Raw_data'/filename
    print(path)
    if filename.split('.')[1]== 'xlsx':
        data = pd.read_excel(path)
    elif filename.split('.')[1]== 'csv':
        data = pd.read_csv(path)
    else:
        'Check file format'
    return data

In [58]:
#Load 5 different excel files 
filename = []
for p in Path( '../Raw_data' ).glob('*.xlsx'):
    print( p.name)
    filename.append(p.name)

for i in range(0,len(filename)):
    exec('{} = load_data(filename[i])'.format(filename[i].split('.')[0]))
    dfname = load_data(filename[i])
    print(filename[i], " loaded in dataframe: ",filename[i].split('.')[0])
    

Telco_customer_churn_population.xlsx
Telco_customer_churn.xlsx
Telco_customer_churn_demographics.xlsx
Telco_customer_churn_services.xlsx
Telco_customer_churn_status.xlsx
Telco_customer_churn_location.xlsx
/Users/disha.dubey/Downloads/dubey031/github/Telco_churn_analysis/Raw_data/Telco_customer_churn_population.xlsx
/Users/disha.dubey/Downloads/dubey031/github/Telco_churn_analysis/Raw_data/Telco_customer_churn_population.xlsx
Telco_customer_churn_population.xlsx  loaded in dataframe:  Telco_customer_churn_population
/Users/disha.dubey/Downloads/dubey031/github/Telco_churn_analysis/Raw_data/Telco_customer_churn.xlsx
/Users/disha.dubey/Downloads/dubey031/github/Telco_churn_analysis/Raw_data/Telco_customer_churn.xlsx
Telco_customer_churn.xlsx  loaded in dataframe:  Telco_customer_churn
/Users/disha.dubey/Downloads/dubey031/github/Telco_churn_analysis/Raw_data/Telco_customer_churn_demographics.xlsx
/Users/disha.dubey/Downloads/dubey031/github/Telco_churn_analysis/Raw_data/Telco_customer_chu

## EDA

In [59]:
Telco_customer_churn_population.info()
Telco_customer_churn.info()
Telco_customer_churn_demographics.info()
Telco_customer_churn_services.info()
Telco_customer_churn_status.info()
Telco_customer_churn_location.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1671 entries, 0 to 1670
Data columns (total 3 columns):
 #   Column      Non-Null Count  Dtype
---  ------      --------------  -----
 0   ID          1671 non-null   int64
 1   Zip Code    1671 non-null   int64
 2   Population  1671 non-null   int64
dtypes: int64(3)
memory usage: 39.3 KB
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7043 entries, 0 to 7042
Data columns (total 33 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   CustomerID         7043 non-null   object 
 1   Count              7043 non-null   int64  
 2   Country            7043 non-null   object 
 3   State              7043 non-null   object 
 4   City               7043 non-null   object 
 5   Zip Code           7043 non-null   int64  
 6   Lat Long           7043 non-null   object 
 7   Latitude           7043 non-null   float64
 8   Longitude          7043 non-null   float64
 9   Gender             7

#### Summary
1. Churn reason and churn category columns have missing values
2. Every table has 7043 rows at customer_id level except for Telco_customer_churn_population table which is at zipcode level


In [62]:
Telco_customer_churn_population.head()
Telco_customer_churn.head()
Telco_customer_churn_demographics.head()
Telco_customer_churn_services.head()
Telco_customer_churn_status.head()
Telco_customer_churn_location.head()

ID  Zip Code  Population
0   1     90001       54492
1   2     90002       44586
2   3     90003       58198
3   4     90004       67852
4   5     90005       43019

CustomerID  Count        Country       State         City  Zip Code  \
0  3668-QPYBK      1  United States  California  Los Angeles     90003   
1  9237-HQITU      1  United States  California  Los Angeles     90005   
2  9305-CDSKC      1  United States  California  Los Angeles     90006   
3  7892-POOKP      1  United States  California  Los Angeles     90010   
4  0280-XJGEX      1  United States  California  Los Angeles     90015   

                 Lat Long   Latitude   Longitude  Gender  ...        Contract  \
0  33.964131, -118.272783  33.964131 -118.272783    Male  ...  Month-to-month   
1   34.059281, -118.30742  34.059281 -118.307420  Female  ...  Month-to-month   
2  34.048013, -118.293953  34.048013 -118.293953  Female  ...  Month-to-month   
3  34.062125, -118.315709  34.062125 -118.315709  Female  ...  Month-to-month   
4  34.039224, -118.266293  34.039224 -118.266293    Male  ...  Month-to-month   

  Paperless Billing             Payment Method  Monthly Charges Total Charges  \
0               Yes               Mailed check            53.85        108.15   
1               Yes           Electronic check            70.70        151.65   
2               Yes           Electronic check            99.65         820.5   
3               Yes           Electronic check           104.80       3046.05   
4               Yes  Bank transfer (automatic)           103.70        5036.3   

  Churn Label Churn Value Churn Score  CLTV                   Churn Reason  
0         Yes           1          86  3239   Competitor made better offer  
1         Yes           1          67  2701                          Moved  
2         Yes           1          86  5372                          Moved  
3         Yes           1          84  5003                          Moved  
4         Yes           1          89  5340  Competitor had better devices  

[5 rows x 33 columns]

Customer ID  Count  Gender  Age Under 30 Senior Citizen Married Dependents  \
0  8779-QRDMV      1    Male   78       No            Yes      No         No   
1  7495-OOKFY      1  Female   74       No            Yes     Yes        Yes   
2  1658-BYGOY      1    Male   71       No            Yes      No        Yes   
3  4598-XLKNJ      1  Female   78       No            Yes     Yes        Yes   
4  4846-WHAFZ      1  Female   80       No            Yes     Yes        Yes   

   Number of Dependents  
0                     0  
1                     1  
2                     3  
3                     1  
4                     1

Customer ID  Count Quarter Referred a Friend  Number of Referrals  \
0  8779-QRDMV      1      Q3                No                    0   
1  7495-OOKFY      1      Q3               Yes                    1   
2  1658-BYGOY      1      Q3                No                    0   
3  4598-XLKNJ      1      Q3               Yes                    1   
4  4846-WHAFZ      1      Q3               Yes                    1   

   Tenure in Months    Offer Phone Service  Avg Monthly Long Distance Charges  \
0                 1     None            No                               0.00   
1                 8  Offer E           Yes                              48.85   
2                18  Offer D           Yes                              11.33   
3                25  Offer C           Yes                              19.76   
4                37  Offer C           Yes                               6.33   

  Multiple Lines  ... Unlimited Data        Contract  Paperless Billing  \
0             No  ...             No  Month-to-Month                Yes   
1            Yes  ...            Yes  Month-to-Month                Yes   
2            Yes  ...            Yes  Month-to-Month                Yes   
3             No  ...            Yes  Month-to-Month                Yes   
4            Yes  ...            Yes  Month-to-Month                Yes   

    Payment Method Monthly Charge Total Charges Total Refunds  \
0  Bank Withdrawal          39.65         39.65          0.00   
1      Credit Card          80.65        633.30          0.00   
2  Bank Withdrawal          95.45       1752.55         45.61   
3  Bank Withdrawal          98.50       2514.50         13.43   
4  Bank Withdrawal          76.50       2868.15          0.00   

  Total Extra Data Charges Total Long Distance Charges Total Revenue  
0                       20                        0.00         59.65  
1                        0                      390.80       1024.10  
2                        0                      203.94       1910.88  
3                        0                      494.00       2995.07  
4                        0                      234.21       3102.36  

[5 rows x 30 columns]

Customer ID  Count Quarter  Satisfaction Score Customer Status Churn Label  \
0  8779-QRDMV      1      Q3                   3         Churned         Yes   
1  7495-OOKFY      1      Q3                   3         Churned         Yes   
2  1658-BYGOY      1      Q3                   2         Churned         Yes   
3  4598-XLKNJ      1      Q3                   2         Churned         Yes   
4  4846-WHAFZ      1      Q3                   2         Churned         Yes   

   Churn Value  Churn Score  CLTV   Churn Category  \
0            1           91  5433       Competitor   
1            1           69  5302       Competitor   
2            1           81  3179       Competitor   
3            1           88  5337  Dissatisfaction   
4            1           67  2793            Price   

                   Churn Reason  
0  Competitor offered more data  
1  Competitor made better offer  
2  Competitor made better offer  
3     Limited range of services  
4            Extra data charges

Location ID Customer ID  Count        Country       State         City  \
0  OXCZEW7397  8779-QRDMV      1  United States  California  Los Angeles   
1  FCCECI8494  7495-OOKFY      1  United States  California  Los Angeles   
2  HEHUQY7254  1658-BYGOY      1  United States  California  Los Angeles   
3  WIUHRF2613  4598-XLKNJ      1  United States  California    Inglewood   
4  CFEZBF4415  4846-WHAFZ      1  United States  California     Whittier   

   Zip Code                Lat Long   Latitude   Longitude  
0     90022   34.02381, -118.156582  34.023810 -118.156582  
1     90063  34.044271, -118.185237  34.044271 -118.185237  
2     90065  34.108833, -118.229715  34.108833 -118.229715  
3     90303  33.936291, -118.332639  33.936291 -118.332639  
4     90602  33.972119, -118.020188  33.972119 -118.020188

In [63]:
Telco_customer_churn_population.describe()
Telco_customer_churn.describe()
Telco_customer_churn_demographics.describe()
Telco_customer_churn_services.describe()
Telco_customer_churn_status.describe()
Telco_customer_churn_location.describe()

ID      Zip Code     Population
count  1671.000000   1671.000000    1671.000000
mean    836.000000  93678.992220   20276.384201
std     482.520466   1817.763591   20689.117300
min       1.000000  90001.000000      11.000000
25%     418.500000  92269.000000    1789.000000
50%     836.000000  93664.000000   14239.000000
75%    1253.500000  95408.000000   32942.500000
max    1671.000000  96161.000000  105285.000000

Count      Zip Code     Latitude    Longitude  Tenure Months  \
count  7043.0   7043.000000  7043.000000  7043.000000    7043.000000   
mean      1.0  93521.964646    36.282441  -119.798880      32.371149   
std       0.0   1865.794555     2.455723     2.157889      24.559481   
min       1.0  90001.000000    32.555828  -124.301372       0.000000   
25%       1.0  92102.000000    34.030915  -121.815412       9.000000   
50%       1.0  93552.000000    36.391777  -119.730885      29.000000   
75%       1.0  95351.000000    38.224869  -118.043237      55.000000   
max       1.0  96161.000000    41.962127  -114.192901      72.000000   

       Monthly Charges  Churn Value  Churn Score         CLTV  
count      7043.000000  7043.000000  7043.000000  7043.000000  
mean         64.761692     0.265370    58.699418  4400.295755  
std          30.090047     0.441561    21.525131  1183.057152  
min          18.250000     0.000000     5.000000  2003.000000  
25%          35.500000     0.000000    40.000000  3469.000000  
50%          70.350000     0.000000    61.000000  4527.000000  
75%          89.850000     1.000000    75.000000  5380.500000  
max         118.750000     1.000000   100.000000  6500.000000

Count          Age  Number of Dependents
count  7043.0  7043.000000           7043.000000
mean      1.0    46.509726              0.468692
std       0.0    16.750352              0.962802
min       1.0    19.000000              0.000000
25%       1.0    32.000000              0.000000
50%       1.0    46.000000              0.000000
75%       1.0    60.000000              0.000000
max       1.0    80.000000              9.000000

Count  Number of Referrals  Tenure in Months  \
count  7043.0          7043.000000       7043.000000   
mean      1.0             1.951867         32.386767   
std       0.0             3.001199         24.542061   
min       1.0             0.000000          1.000000   
25%       1.0             0.000000          9.000000   
50%       1.0             0.000000         29.000000   
75%       1.0             3.000000         55.000000   
max       1.0            11.000000         72.000000   

       Avg Monthly Long Distance Charges  Avg Monthly GB Download  \
count                        7043.000000              7043.000000   
mean                           22.958954                20.515405   
std                            15.448113                20.418940   
min                             0.000000                 0.000000   
25%                             9.210000                 3.000000   
50%                            22.890000                17.000000   
75%                            36.395000                27.000000   
max                            49.990000                85.000000   

       Monthly Charge  Total Charges  Total Refunds  Total Extra Data Charges  \
count     7043.000000    7043.000000    7043.000000               7043.000000   
mean        64.761692    2280.381264       1.962182                  6.860713   
std         30.090047    2266.220462       7.902614                 25.104978   
min         18.250000      18.800000       0.000000                  0.000000   
25%         35.500000     400.150000       0.000000                  0.000000   
50%         70.350000    1394.550000       0.000000                  0.000000   
75%         89.850000    3786.600000       0.000000                  0.000000   
max        118.750000    8684.800000      49.790000                150.000000   

       Total Long Distance Charges  Total Revenue  
count                  7043.000000    7043.000000  
mean                    749.099262    3034.379056  
std                     846.660055    2865.204542  
min                       0.000000      21.360000  
25%                      70.545000     605.610000  
50%                     401.440000    2108.640000  
75%                    1191.100000    4801.145000  
max                    3564.720000   11979.340000

Count  Satisfaction Score  Churn Value  Churn Score         CLTV
count  7043.0         7043.000000  7043.000000  7043.000000  7043.000000
mean      1.0            3.244924     0.265370    58.505040  4400.295755
std       0.0            1.201657     0.441561    21.170031  1183.057152
min       1.0            1.000000     0.000000     5.000000  2003.000000
25%       1.0            3.000000     0.000000    40.000000  3469.000000
50%       1.0            3.000000     0.000000    61.000000  4527.000000
75%       1.0            4.000000     1.000000    75.500000  5380.500000
max       1.0            5.000000     1.000000    96.000000  6500.000000

Count      Zip Code     Latitude    Longitude
count  7043.0   7043.000000  7043.000000  7043.000000
mean      1.0  93486.071134    36.197455  -119.756684
std       0.0   1856.768045     2.468929     2.154425
min       1.0  90001.000000    32.555828  -124.301372
25%       1.0  92101.000000    33.990646  -121.788090
50%       1.0  93518.000000    36.205465  -119.595293
75%       1.0  95329.000000    38.161321  -117.969795
max       1.0  96150.000000    41.962127  -114.192901

## Join all the datasets to form a master dataset

In [67]:
Telco_customer_churn_population.columns
Telco_customer_churn.columns
Telco_customer_churn_demographics.columns
Telco_customer_churn_services.columns
Telco_customer_churn_status.columns
Telco_customer_churn_location.columns

Index(['ID', 'Zip Code', 'Population'], dtype='object')

Index(['CustomerID', 'Count', 'Country', 'State', 'City', 'Zip Code',
       'Lat Long', 'Latitude', 'Longitude', 'Gender', 'Senior Citizen',
       'Partner', 'Dependents', 'Tenure Months', 'Phone Service',
       'Multiple Lines', 'Internet Service', 'Online Security',
       'Online Backup', 'Device Protection', 'Tech Support', 'Streaming TV',
       'Streaming Movies', 'Contract', 'Paperless Billing', 'Payment Method',
       'Monthly Charges', 'Total Charges', 'Churn Label', 'Churn Value',
       'Churn Score', 'CLTV', 'Churn Reason'],
      dtype='object')

Index(['Customer ID', 'Count', 'Gender', 'Age', 'Under 30', 'Senior Citizen',
       'Married', 'Dependents', 'Number of Dependents'],
      dtype='object')

Index(['Customer ID', 'Count', 'Quarter', 'Referred a Friend',
       'Number of Referrals', 'Tenure in Months', 'Offer', 'Phone Service',
       'Avg Monthly Long Distance Charges', 'Multiple Lines',
       'Internet Service', 'Internet Type', 'Avg Monthly GB Download',
       'Online Security', 'Online Backup', 'Device Protection Plan',
       'Premium Tech Support', 'Streaming TV', 'Streaming Movies',
       'Streaming Music', 'Unlimited Data', 'Contract', 'Paperless Billing',
       'Payment Method', 'Monthly Charge', 'Total Charges', 'Total Refunds',
       'Total Extra Data Charges', 'Total Long Distance Charges',
       'Total Revenue'],
      dtype='object')

Index(['Customer ID', 'Count', 'Quarter', 'Satisfaction Score',
       'Customer Status', 'Churn Label', 'Churn Value', 'Churn Score', 'CLTV',
       'Churn Category', 'Churn Reason'],
      dtype='object')

Index(['Location ID', 'Customer ID', 'Count', 'Country', 'State', 'City',
       'Zip Code', 'Lat Long', 'Latitude', 'Longitude'],
      dtype='object')

In [97]:
#merge1 = pd.merge(Telco_customer_churn,Telco_customer_churn_demographics, left_on = 'CustomerID',right_on = 'Customer ID', how = 'left')
merge1 = pd.merge(Telco_customer_churn_demographics,Telco_customer_churn_services, on = 'Customer ID', how = 'inner')
merge2 = pd.merge(Telco_customer_churn_status,Telco_customer_churn_location, on = 'Customer ID', how = 'inner')
merge3 = pd.merge(merge1,merge2, on = 'Customer ID', how = 'inner')
all_combined = pd.merge(merge3,Telco_customer_churn_population, on = 'Zip Code', how = 'left')


In [98]:
all_combined.head()

Customer ID  Count_x_x  Gender  Age Under 30 Senior Citizen Married  \
0  8779-QRDMV          1    Male   78       No            Yes      No   
1  7495-OOKFY          1  Female   74       No            Yes     Yes   
2  1658-BYGOY          1    Male   71       No            Yes      No   
3  4598-XLKNJ          1  Female   78       No            Yes     Yes   
4  4846-WHAFZ          1  Female   80       No            Yes     Yes   

  Dependents  Number of Dependents  Count_y_x  ... Count_y_y        Country  \
0         No                     0          1  ...         1  United States   
1        Yes                     1          1  ...         1  United States   
2        Yes                     3          1  ...         1  United States   
3        Yes                     1          1  ...         1  United States   
4        Yes                     1          1  ...         1  United States   

        State         City Zip Code                Lat Long   Latitude  \
0  California  Los Angeles    90022   34.02381, -118.156582  34.023810   
1  California  Los Angeles    90063  34.044271, -118.185237  34.044271   
2  California  Los Angeles    90065  34.108833, -118.229715  34.108833   
3  California    Inglewood    90303  33.936291, -118.332639  33.936291   
4  California     Whittier    90602  33.972119, -118.020188  33.972119   

    Longitude   ID Population  
0 -118.156582   21      68701  
1 -118.185237   54      55668  
2 -118.229715   56      47534  
3 -118.332639  100      27778  
4 -118.020188  114      26265  

[5 rows x 59 columns]

In [99]:
all_combined.columns

Index(['Customer ID', 'Count_x_x', 'Gender', 'Age', 'Under 30',
       'Senior Citizen', 'Married', 'Dependents', 'Number of Dependents',
       'Count_y_x', 'Quarter_x', 'Referred a Friend', 'Number of Referrals',
       'Tenure in Months', 'Offer', 'Phone Service',
       'Avg Monthly Long Distance Charges', 'Multiple Lines',
       'Internet Service', 'Internet Type', 'Avg Monthly GB Download',
       'Online Security', 'Online Backup', 'Device Protection Plan',
       'Premium Tech Support', 'Streaming TV', 'Streaming Movies',
       'Streaming Music', 'Unlimited Data', 'Contract', 'Paperless Billing',
       'Payment Method', 'Monthly Charge', 'Total Charges', 'Total Refunds',
       'Total Extra Data Charges', 'Total Long Distance Charges',
       'Total Revenue', 'Count_x_y', 'Quarter_y', 'Satisfaction Score',
       'Customer Status', 'Churn Label', 'Churn Value', 'Churn Score', 'CLTV',
       'Churn Category', 'Churn Reason', 'Location ID', 'Count_y_y', 'Country',
       'Stat

In [100]:
all_combined.drop(['Count_x_x','Count_y_x','Count_x_y', 'Count_x_y','Quarter_x','Quarter_y','Count_y_y','ID'], axis = 1,inplace = True)

In [86]:
all_combined.head()

Customer ID  Gender  Age Under 30 Senior Citizen Married Dependents  \
0  8779-QRDMV    Male   78       No            Yes      No         No   
1  7495-OOKFY  Female   74       No            Yes     Yes        Yes   
2  1658-BYGOY    Male   71       No            Yes      No        Yes   
3  4598-XLKNJ  Female   78       No            Yes     Yes        Yes   
4  4846-WHAFZ  Female   80       No            Yes     Yes        Yes   

   Number of Dependents Quarter Referred a Friend  ...  \
0                     0      Q3                No  ...   
1                     1      Q3               Yes  ...   
2                     3      Q3                No  ...   
3                     1      Q3               Yes  ...   
4                     1      Q3               Yes  ...   

                   Churn Reason  Location ID        Country       State  \
0  Competitor offered more data   OXCZEW7397  United States  California   
1  Competitor made better offer   FCCECI8494  United States  California   
2  Competitor made better offer   HEHUQY7254  United States  California   
3     Limited range of services   WIUHRF2613  United States  California   
4            Extra data charges   CFEZBF4415  United States  California   

          City Zip Code                Lat Long   Latitude   Longitude  \
0  Los Angeles    90022   34.02381, -118.156582  34.023810 -118.156582   
1  Los Angeles    90063  34.044271, -118.185237  34.044271 -118.185237   
2  Los Angeles    90065  34.108833, -118.229715  34.108833 -118.229715   
3    Inglewood    90303  33.936291, -118.332639  33.936291 -118.332639   
4     Whittier    90602  33.972119, -118.020188  33.972119 -118.020188   

  Population  
0      68701  
1      55668  
2      47534  
3      27778  
4      26265  

[5 rows x 53 columns]

In [103]:
all_combined.describe(include = 'all')

Customer ID Gender          Age Under 30 Senior Citizen Married  \
count         7043   7043  7043.000000     7043           7043    7043   
unique        7043      2          NaN        2              2       2   
top     8779-QRDMV   Male          NaN       No             No      No   
freq             1   3555          NaN     5642           5901    3641   
mean           NaN    NaN    46.509726      NaN            NaN     NaN   
std            NaN    NaN    16.750352      NaN            NaN     NaN   
min            NaN    NaN    19.000000      NaN            NaN     NaN   
25%            NaN    NaN    32.000000      NaN            NaN     NaN   
50%            NaN    NaN    46.000000      NaN            NaN     NaN   
75%            NaN    NaN    60.000000      NaN            NaN     NaN   
max            NaN    NaN    80.000000      NaN            NaN     NaN   

       Dependents  Number of Dependents Referred a Friend  \
count        7043           7043.000000              7043   
unique          2                   NaN                 2   
top            No                   NaN                No   
freq         5416                   NaN              3821   
mean          NaN              0.468692               NaN   
std           NaN              0.962802               NaN   
min           NaN              0.000000               NaN   
25%           NaN              0.000000               NaN   
50%           NaN              0.000000               NaN   
75%           NaN              0.000000               NaN   
max           NaN              9.000000               NaN   

        Number of Referrals  ...                   Churn Reason Location ID  \
count           7043.000000  ...                           1869        7043   
unique                  NaN  ...                             20        7043   
top                     NaN  ...  Competitor had better devices  OXCZEW7397   
freq                    NaN  ...                            313           1   
mean               1.951867  ...                            NaN         NaN   
std                3.001199  ...                            NaN         NaN   
min                0.000000  ...                            NaN         NaN   
25%                0.000000  ...                            NaN         NaN   
50%                0.000000  ...                            NaN         NaN   
75%                3.000000  ...                            NaN         NaN   
max               11.000000  ...                            NaN         NaN   

              Country       State         City      Zip Code  \
count            7043        7043         7043   7043.000000   
unique              1           1         1106           NaN   
top     United States  California  Los Angeles           NaN   
freq             7043        7043          293           NaN   
mean              NaN         NaN          NaN  93486.071134   
std               NaN         NaN          NaN   1856.768045   
min               NaN         NaN          NaN  90001.000000   
25%               NaN         NaN          NaN  92101.000000   
50%               NaN         NaN          NaN  93518.000000   
75%               NaN         NaN          NaN  95329.000000   
max               NaN         NaN          NaN  96150.000000   

                      Lat Long     Latitude    Longitude     Population  
count                     7043  7043.000000  7043.000000    7043.000000  
unique                    1679          NaN          NaN            NaN  
top     33.362575, -117.299644          NaN          NaN            NaN  
freq                        43          NaN          NaN            NaN  
mean                       NaN    36.197455  -119.756684   22139.814568  
std                        NaN     2.468929     2.154425   21152.174407  
min                        NaN    32.555828  -124.301372      11.000000  
25%                        NaN    33.990646  -121.788090    2344.000000  
5

In [105]:
# remove special character
all_combined.columns = all_combined.columns.str.replace(' ', '_')

In [106]:
all_combined.to_csv('../Raw_data/master_dataset.csv', index = False)

In [107]:
pwd

'/Users/disha.dubey/Downloads/dubey031/github/Telco_churn_analysis/Code'